core/src/main/java/com/netease/arctic/io/reader/CombinedDeleteFilter.java

```diff
@@ -121,15 +138,40 @@ protected CombinedDeleteFilter(
         }
       }
     }
-
-    this.positionPathSets = positionPathSets;
+    this.positionPathSets =
+        Arrays.stream(rewriteFilesInput.dataFiles())
+            .map(s -> s.path().toString())
+            .collect(Collectors.toSet());
     this.posDeletes = posDeleteBuilder.build();
     this.eqDeletes = eqDeleteBuilder.build();
     this.deleteSchema = TypeUtil.select(tableSchema, deleteIds);
 
     if (structLikeCollections != null) {
       this.structLikeCollections = structLikeCollections;
     }
+    this.filterEqDelete = filterEqDelete();
+  }
+
+  /**
+   * Whether to use {@link BloomFilter} to filter eq delete and reduce the amount of data written to
+   * {@link StructLikeBaseMap} by eq delete
+   */
+  private boolean filterEqDelete() {
+    long dataRecordCnt = Arrays.stream(input.dataFiles()).mapToLong(ContentFile::recordCount).sum();
+    long eqDeleteRecordCnt =
+        Arrays.stream(input.deleteFiles())
+            .filter(file -> file.content() == FileContent.EQUALITY_DELETES)
+            .mapToLong(ContentFile::recordCount)
+            .sum();
+
+    // Data file primary key memory usage, cost is the same as eq delete, +1 cost
+    // Data file read cost is 0.5 times the memory cost, +0.5 cost
+    return eqDeleteRecordCnt > dataRecordCnt * 2.5;
```

Comment:
We should set a minimum number of records for equality delete to prevent triggering this optimization logic even when the data volume is relatively small.

Reply:
Fixed.